### ISSUE TITLE
[MU4 Issue] Dots on notes that cross between voices can overlap

### ISSUE THREAD

**Issue by MarcSabatella** [musescore/MuseScore#16210]
**Describe the bug**
In cases where voices cross (voice 2 above 1), dots on notes can overlap

![image](https://user-images.githubusercontent.com/1799936/216760263-8c879550-aeec-42e7-b4f6-09c62e74f065.png)

Those are both single-dotted notes.  When flags are involved, the dots are in the same space and it looks like a double dot. 
 When no flags are involved, the dots overlap each other and you aren't sure which note the dot applies to, or is it both.

**Platform information**
 - OS: Linux

This example is adapted from one encountered by @Jojo-Schmitz.  It's not clear to me what the right answer is.  @oktophonie @mike-spa

**Comment by oktophonie** [comment]
Normally in a voice 2 (or 4) the dot for a note that's on a line will go to the space below, which creates the ambiguous notation you show here.

We actually addressed this special case in the as-yet-unmerged https://github.com/musescore/MuseScore/pull/12547 which moves the dot to the space above in this special context:
![image](https://user-images.githubusercontent.com/20604032/216761043-bc5c746d-b280-47b6-bd0c-6f59f82141a5.png)

Doesn't solve the problem of one off the dots colliding with the stem though, which is a separate thing.

**Comment by filipwerthmann** [comment]
Are there any clear answer and instructions how to fix that bug?

**Comment by oktophonie** [comment]
Not really; this is a very unusual situation. Either of these two solutions ar possible (and would be better than the current mess):
![image](https://user-images.githubusercontent.com/20604032/219370263-49dc9c35-dd68-4ae9-8426-2472a2d8684d.png)
The former is probably preferable, but you can't win here really.

**Comment by filipwerthmann** [comment]
I have kinda same bug/issue? I attach 2 screenshots, the first one is the original and the second is how my score is...

![IMG_20230216_005755](https://user-images.githubusercontent.com/125568335/219371402-0c71f30c-105a-47c4-885b-c4b30063fee4.jpg)
![IMG_20230216_005541_HDR](https://user-images.githubusercontent.com/125568335/219371405-dcab2af0-0dd2-4460-a9d8-848b0791c7fa.jpg)

**Comment by oktophonie** [comment]
Good example. I'll use these to formulate a separate issue, now.
